sqlglot/parser.py

```diff
@@ -2789,6 +2789,12 @@ def _parse_type(self) -> t.Optional[exp.Expression]:
 
         return this
 
+    def _parse_type_size_specifier(self) -> t.Optional[exp.Expression]:
+        this = self._parse_conjunction()
+        if not this:
+            return None
+        return self.expression(exp.DataTypeSizeSpecifier, this=this, expression=self._parse_type())
+
```

Comment:
I think we can make this a bit safer (i.e. backwards compatible) if we return `this`, in case `_parse_type` returns `None`.

Reply:
why was this a conjunction before?

Reply:
Does that mean that INT(x / 10000) is now a data type specifier? that should be fixed

Reply:
Correction: how we parse this depends on the context. Since `INT` is used as a function in this example, it will get parsed as an anonymous one (or as a `Cast` for spark -- see Vegard's recent additions). For example:

```
>>> sqlglot.parse_one("int(x / 100)")
(ANONYMOUS this: int, expressions:
  (DIV this:
    (COLUMN this:
      (IDENTIFIER this: x, quoted: False)), expression:
    (LITERAL this: 100, is_string: False)))
>>> sqlglot.parse_one("cast(x as int(x / 100))")  -- This doesn't make sense
(CAST this:
  (COLUMN this:
    (IDENTIFIER this: x, quoted: False)), to:
  (DATATYPE this: Type.INT, expressions:
    (DATATYPESIZE this:
      (DIV this:
        (COLUMN this:
          (IDENTIFIER this: x, quoted: False)), expression:
        (LITERAL this: 100, is_string: False))), nested: False, prefix: False))
>>> 
```

To conclude, we don't ever reach `_parse_type_size_specifier` in `sqlglot.parse_one("select int(x/100)")`.

Reply:
yea probably, I'm taking a look

Reply:
Seems like we can get rid of it.. all tests seem to pass in main with:
```
-                expressions = self._parse_csv(self._parse_conjunction)
+                expressions = self._parse_csv(self._parse_type)
```

Reply:
I think it's good now, let me know if I can make any other improvements.